src/policy/mempool.cpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2020 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/mempool.h>
+
+#include <validation/forks.h>
+
+uint32_t GetBCHDefaultAncestorLimit(const Consensus::Params &params, const CBlockIndex *pindexPrev)
+{
+    return IsMay2020Enabled(params, pindexPrev) ? BCH_DEFAULT_ANCESTOR_LIMIT_LONGER
+                                                : BCH_DEFAULT_ANCESTOR_LIMIT;
+}
+
+uint32_t GetBCHDefaultDescendantLimit(const Consensus::Params &params, const CBlockIndex *pindexPrev)
```

Comment:
As written the limit won't be bumped at fork time.  It will only be bumped if a node is connected after fork time.  Whenever a block is added, we need to see if May2020 activated and then if so update the values in the CNode array.

Reply:
Maybe I missing something but `IsMay2020Enabled` is a wrapper around:

```c++
/** This will check if the Fork will be enabled at the next block
 * i.e. we are at block x - 1, [x-1, +inf]
 * state fork: enabled or activated */
bool CBlockIndex::IsforkActiveOnNextBlock(int time) const
{
    if (time == 0)
        return false;
    // if the fork is already activated
    if (forkActivated(time))
        return true;
    if (GetMedianTimePast() >= time)
        return true;
    return false;
}
```

where time is MTP of current tip. 

So it returns the new limit for every block that belongs to the interval that goes from the activation block and on, since we are using `GetBCHDefaultAncestorLimit()` directly in `txadmission` (line 1155) I think it should behave as expected other wise also `test/mempool_policy_tests.cpp` should have failed.

Reply:
you are correct, but we need to apply to different checks. one for bu pers that communicate their limit via xversion and one for default BCH node that don;t use xversion (ABC, BCHN, etc etc) for which we assume that the limit is going to be bumped at upgrade time. 

Just pushed the code that implement that. 

Using only the new getters here would mean to use default BCH policy also for BU nodes that use xversion to communicate the limits and this would lead to a failure of (at least) `mempool_push.py`

```
- stderr --------------------------------------------------
   File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/test_framework.py", line 271, in main
    self.run_test()
  File "/home/sickpig/src/bu-sickpig/build-bump50/../qa/rpc-tests/mempool_push.py", line 168, in run_test
    waitFor(DELAY_TIME, lambda: self.nodes[1].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH*2)
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/util.py", line 140, in waitFor
```